plugin/pkg/admission/noderestriction/admission.go

```diff
@@ -369,6 +372,31 @@ func resourceClaimStatusesEqual(statusA, statusB []api.PodResourceClaimStatus) b
 	return true
 }
 
+func extendedResourceClaimStatusEqual(statusA, statusB *api.PodExtendedResourceClaimStatus) bool {
+	if statusA == nil && statusB == nil {
+		return true
+	}
+	if statusA == nil || statusB == nil {
+		return false
+	}
+	if statusA.ResourceClaimName != statusB.ResourceClaimName {
+		return false
+	}
+	if len(statusA.RequestMappings) != len(statusB.RequestMappings) {
+		return false
+	}
+
+	// In most cases, status entries only get added once and not modified.
+	// But this cannot be guaranteed, so for the sake of correctness in all
+	// cases this code here has to check.
+	for i := range statusA.RequestMappings {
+		if statusA.RequestMappings[i] != statusB.RequestMappings[i] {
+			return false
+		}
+	}
+	return true
```

Comment:
You can actually try:
```suggestion
	return slices.Equal(statusA.RequestMapping, statusB.RequestMapping)
```

Reply:
And also remove the length check if using this method.

Reply:
done